website/docs/concepts/function-components/pure-components.mdx

```diff
@@ -2,15 +2,12 @@
 title: 'Pure Components'
 ---
 
-## Pure function definition
+A function component is considered [pure] when the returned `Html` is deterministically derived
+from its props, and its view function mutates no state or has other side-effects.
 
-A function is considered pure when the return values are deterministic given the same parameters, and it has no side-effects.
+[pure]: https://en.wikipedia.org/wiki/Pure_function
 
-## Pure components
-
-In the same sense component is pure when it does not have any state or side_effects and with the given props always returns the same `Html`.
-
-For example below is a pure component, for a given prop `is_loading` it will always result in same `Html` without any side effects.
+For example below is a pure component. For a given prop `is_loading` it will always result in the same `Html` without any side effects.
```

Comment:
We should add a note that a function returning `Html` should preferred over a component. Rendering a tag is less expensive than a component. Pure components are just relying on the optimizer to optimize out all the extra machinery that's needed for components.

Reply:
Do note that using hooks does not automatically make a component impure and is not possible in "just" a function.

Reply:
True, but if it's possible to skip all the component-related code in the lifecycle, we should advise to skip it

Reply:
It can actually make sense to wrap stuff in components, since the reconciliation on a new render has to do less work, if the nested components don't change. Explaining that trade-off early on in the docs is a bit harder to do.